### PR TITLE
feat: Add ability to run esrallyd inside docker [ES-9146]

### PR DESCRIPTION
**Pull request by favilo**
This waits for child processes to exit before exiting itself. This is necessary, because `docker -d` will just terminate the container once the main process ends. 

Also adding ability to log json files.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -84,6 +84,8 @@ dependencies = [
     "zstandard==0.21.0",
     # License: Python Software Foundation License
     "typing-extensions==4.12.2",
+    # License:  BSD-2-Clause license 
+    "python-json-logger==2.0.7",
```
Comment (favilo) [inline pyproject.toml:88]: This isn't used directly, but it allows us to specify JSON logs in our `logging.json` configuration. Useful if you are using docker, and can't add dependencies in your virtual environment.

See change in [configuration.rst] for example of how to do this.

File: esrally/utils/process.py [hunk 2]
```diff
@@ -229,3 +229,35 @@ def rally_process(p: psutil.Process) -> bool:
...
+) -> bool:
+    """
+    Waits for all child processes to terminate.
+
+    :param timeout: The maximum time to wait for child processes to terminate (default: None).
+    :param callback: A callback to call as each child process terminates.
+        The callback will be passed the PID and the return code of the child process.
+    :param list_callback: A callback to tell caller about the child processes that are being waited for.
+
+    :return: True if no child processes found, False otherwise.
```
Comment (dpifke-elastic) [inline esrally/utils/process.py:254]: This seems backwards to the actual behavior.

File: esrally/utils/process.py [hunk 3]
```diff
@@ -229,3 +229,35 @@ def rally_process(p: psutil.Process) -> bool:
         )
 
     kill_all(rally_process)
+
+
+def current_pid() -> int:
```
Comment (dpifke-elastic) [inline esrally/utils/process.py:234]: Why do we need to wrap os.getpid?

**Review by dpifke-elastic — APPROVED** [review]
LGTM, just one (maybe dumb) question and a doc update.

File: esrally/utils/process.py (hunk 3 above)
Comment (favilo) [reply esrally/utils/process.py]: Heh, this was left over from when I was using `psutil` for the purpose, but realized this was a much faster call.

**Pull request merged**